include/wpe/gamepad.h

```diff
@@ -0,0 +1,285 @@
+/*
+ * If not stated otherwise in this file or this component's Licenses.txt file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2020 RDK Management
+ * Copyright 2021 Igalia, S.L.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if !defined(__WPE_H_INSIDE__) && !defined(WPE_COMPILATION)
+#error "Only <wpe/wpe.h> can be included directly."
+#endif
+
+#ifndef wpe_gamepad_h
+#define wpe_gamepad_h
+
+#if defined(WPE_COMPILATION)
+#include <wpe/export.h>
+#endif
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Standard mapping
+// Refer https://www.w3.org/TR/gamepad/#gamepadbutton-interface
+enum StandardGamepadAxis {
+    GAMEPAD_AXIS_UNKNOWN = -1,
+    GAMEPAD_AXIS_LEFT_STICK_X,
+    GAMEPAD_AXIS_LEFT_STICK_Y,
+    GAMEPAD_AXIS_RIGHT_STICK_X,
+    GAMEPAD_AXIS_RIGHT_STICK_Y,
+    GAMEPAD_AXIS_COUNT,
+};
+
+enum StandardGamepadButton {
```

Comment:
Same here, let's use `enum wpe_gamepad_button` + `WPE_GAMEPAD_BUTTON_*`.

As above, is the `_BUTTON_UNKNOWN` really needed?

Lastly, button names `A`, `B`, `X`, and `Y` seem pretty much tied to the names given by a certain company where games feature Italian plumber twins. The W3C spec calls them “bottom button in right cluster”, which is a mouthful but definitely more generic; I think we could use `_BUTTON_BOTTOM`, `_BUTTON_RIGHT`, etc. There is no ambiguity with the left button cluster because this other one contains the directional buttons which contain the `D_PAD` string in their names.